lisa/target.py

```diff
@@ -629,9 +633,9 @@ def _init_target(self, kind, name, workdir, device, host,
                 device = 'DEFAULT'
 
             conn_settings['device'] = device
-            # If the username was explicitly set to "root", root the target as
-            # early as possible
-            conn_settings['adb_as_root'] = (username == 'root')
+            # If the username was explicitly set to "root", or username is None,
+            # root the target as early as possible
+            conn_settings['adb_as_root'] = (username == 'root' or username is None)
```

Comment:
only `username=='root'` should trigger the "mandatory" rooting of the device, otherwise non-rooted device will not be usable anymore.